tiledb/sm/filesystem/s3.cc

```diff
@@ -1492,20 +1508,11 @@ Status S3::init_client() const {
   {
     std::lock_guard<std::mutex> static_lck(static_client_init_mtx);
     if (credentials_provider_ == nullptr) {
-      client_ = make_shared<TileDBS3Client>(
-          HERE(),
-          s3_params_,
-          *client_config_,
-          Aws::Client::AWSAuthV4Signer::PayloadSigningPolicy::Never,
-          s3_params_.use_virtual_addressing_);
+      client_ =
+          make_shared<TileDBS3Client>(HERE(), s3_params_, *client_config_);
     } else {
       client_ = make_shared<TileDBS3Client>(
-          HERE(),
-          s3_params_,
-          credentials_provider_,
-          *client_config_,
-          Aws::Client::AWSAuthV4Signer::PayloadSigningPolicy::Never,
-          s3_params_.use_virtual_addressing_);
+          HERE(), s3_params_, credentials_provider_, *client_config_);
     }
```

Comment:
We can unify the two constructors and create a `DefaultAWSCredentialsProviderChain` ourselves if the credentials provider is null. I haven't been the only person who was surprised when learning that the SDK created a default provider if you did not specify it.